### PR TITLE
Make /decide endpoint more flexible

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

*Please describe.*  
*If this affects the front-end, include screenshots.*  

Our /decide endpoint only accepted base64 data, as well as looked for the API key under "token".

It now accepts both base64 and non-base64 data, while keeping the functionality intact for our JS integration. It also accepts "api_key" as a key like our other endpoints.

## Checklist

- [ ] All querysets/queries filter by Team (if this PR affects any querysets/queries)
- [ ] Backend tests (if this PR affects the backend)
- [ ] Cypress E2E tests (if this PR affects the front and/or backend)

### CODE REVIEW

**Comment by yakkomajuri** [comment]
Note to self: Our Cypress tests really need fixing

File: posthog/api/decide.py [hunk 1]
```diff
@@ -12,21 +14,73 @@
 from posthog.utils import cors_response
 
 
-def _load_data(data: str) -> Dict[str, Any]:
+def _load_data(request) -> Optional[Dict[str, Any]]:
```
Comment (timgl) [inline posthog/api/decide.py:17]: This function is the exact same as in capture.py. Could be worth separating it out into utils.py

Comment (yakkomajuri) [reply posthog/api/decide.py]: Done!

**Comment by yakkomajuri** [comment]
@timgl In addition to moving load_data over to utils.py, I also configured the /decide endpoint to allow Personal API Key authentication just like /capture. Spoke to @Twixes and this is something that just went under the radar on the API Key PR. 

This will make things much easier for people since they won't need to juggle 2 keys to use feature flags. At the moment, you can only create and edit flags with the Personal Key, and can only get data on flags with the Team Key.

Now both can be done with the Personal API Key.

**Comment by Twixes** [comment]
This looks pretty good, I just added some tests for this under TestPersonalAPIKeysAPIAuthentication and the one for `/decide` fails, as the endpoint returns isAuthenticated false. I'd think it should return true, but not entirely sure about this.

**Comment by yakkomajuri** [comment]
Huh, odd. I ran curl requests and everything seemed fine. Will look into this when I get the chance.

**Comment by yakkomajuri** [comment]
@Twixes I gave it a very quick look. `isAuthenticated` is a Toolbar thing, so I believe  it's not actually testing if the key worked for authentication, just if the Toolbar can run on a given webpage and that the user sending the request can see it

**Comment by Twixes** [comment]
Let's get this in Monday maybe @yakkomajuri? #1581 appeared, so I'm wondering how this will work together.

**Comment by yakkomajuri** [comment]
@Twixes yeah definitely - I got caught up with the release so this got behind. I'll sort it out and determine what to do about the other PR. Shouldn't be a big hassle anyway, it's just a minor change.

**Comment by yakkomajuri** [comment]
This is weird. Tests pass locally for me but the error on the PR check is:

`kombu.exceptions.OperationalError: Error 111 connecting to localhost:6379. Connection refused.`

**Comment by yakkomajuri** [comment]
Closing in favor of #1592

**Pull request closed without merging**